### ISSUE TITLE
Worksheet 3 - Karpathy switches to Pytorch cross_entropy() for loss

### ISSUE THREAD

**Issue by Russ741** [Russ741/karpathy-nn-z2h#19]
At present I have get_prob and get_loss functions spelled out explicitly. Karpathy switches to using torch.nn.functional.cross_entropy for his loss calculation (video: 0:32:49).

get_prob is called in forward_prop.
forward_prop is called in train_once, get_distribution, and get_empty_word_prob.
get_loss is called in train_once.

I think the best solution is to return logits from forward_prop.
  train_once: call cross_entropy with the returned logits
  get_distribution: call get_prob with the results of forward_prop
  get_empty_word_prob: call get_prob with the results of forward_prop

Other options are to have forward_prop return loss, but this would make it harder to generate words, etc.

**Comment by Russ741** [comment]
Karpathy also uses torch.nn.functional.softmax for his probability distribution calculation rather than doing it manually (video: https://youtu.be/TCH_1BHY58I?t=4404).
